packages/realm_dart/CMakePresets.json

```diff
@@ -168,7 +168,7 @@
       "configurePreset": "macos",
       "nativeToolOptions": [
         "-destination",
-        "platform=macOS"
+        "generic/platform=macOS"
```

Comment:
As of XCode 15.3 it is apparently important to use `generic/platform=macOS` during preset, to actually end up with a universal binary (ie. supporting both `arm64` and `x86_64`).

The downside of this change is that we also build for `x86_64` on our dev machines, but given that we use `ccache` it is perhaps a non-issue in practice.